components/message/index.tsx

```diff
@@ -54,25 +64,36 @@ function notice(
   }
 
   const target = key++;
-  getMessageInstance((instance) => {
-    instance.notice({
-      key: target,
-      duration,
-      style: {},
-      content: (
-        <div className={`${prefixCls}-custom-content ${prefixCls}-${type}`}>
-          <Icon type={iconType} />
-          <span>{content}</span>
-        </div>
-      ),
-      onClose,
+  const closePromise = new Promise((resolve) => {
```

Comment:
https://caniuse.com/#search=Promise

Reply:
I am not sure.  Do you mean some browsers ant-design supporting does not support `Promise`?

Reply:
I think it's ok, never mind.